### ISSUE TITLE
[Bug]: Kubernetes deployment not using hostname from /etc/hostname

### ISSUE THREAD

**Issue by bdkuhman** [netdata/netdata#16396]
### Bug description

/etc/hostname is overridden with extraVolumeMounts/extraVolumes in helm install, but hostnames continue to be pod names.

### Expected behavior

node hostnames should be read from /etc/hostname

### Steps to reproduce

Override helm chart with following values:
```
child:
    hostNetwork: false
    dnsPolicy: ClusterFirst
    extraVolumeMounts:
        - mountPath: /etc/hostname
          name: hostname
    extraVolumes:
        - name: hostname
          hostPath:
            path: /etc/hostname
            type: File
            readOnly: true
```

### Installation method

helmchart (kubernetes)

### System info

```shell
Linux netdata-child-m2lxz 6.1.43-08467-g9b183cb29b85 #1 SMP PREEMPT_DYNAMIC Thu Aug  3 19:55:27 PDT 2023 aarch64 Linux
/etc/alpine-release:3.18.4
/etc/os-release:NAME="Alpine Linux"
/etc/os-release:ID=alpine
/etc/os-release:VERSION_ID=3.18.4
/etc/os-release:PRETTY_NAME="Alpine Linux v3.18"
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.43.2
    Installation Type __________________________________________ : oci
    Package Architecture _______________________________________ : aarch64
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ :  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-math' '--with-user=netdata' '--without-bundled-protobuf' '--disable-dependency-tracking' '--enable-lto' '--disable-ebpf' 'CFLAGS=-ffunction-sectio
ns -fdata-sections -O2 -funroll-loops -pipe' 'LDFLAGS=-Wl,--gc-sections'
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.1.43-08467-g9b183cb29b85
    Operating System ___________________________________________ : Debian GNU/Linux
    Operating System ID ________________________________________ : debian
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : 11 (bullseye)
    Operating System Version ID ________________________________ : 3.18.4
    Detection __________________________________________________ : /host/etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 4
    CPU Frequency ______________________________________________ : 1416000000
    CPU Architecture ___________________________________________ : 2043523072
    RAM Bytes __________________________________________________ : 251454816256
    Disk Capacity ______________________________________________ : aarch64
    Virtualization Technology __________________________________ : unknown
    Virtualization Detection ___________________________________ : none
Container:
    Container __________________________________________________ : container
    Container Detection ________________________________________ : kubernetes
    Container Orchestrator _____________________________________ : kubernetes
    Container Operating System _________________________________ : Alpine Linux
    Container Operating System ID ______________________________ : alpine
    Container Operating System ID Like _________________________ : unknown
    Container Operating System Version _________________________ : unknown
    Container Operating System Version ID ______________________ : 3.18.4
    Container Operating System Detection _______________________ : /etc/os-release
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (lz4)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine ___________________________________________________ : YES
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    map ________________________________________________________ : YES
    save _______________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : NO
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Judy (high-performance dynamic arrays and hashtables) ______ : YES (bundled)
    dlib (robust machine learning toolkit) _____________________ : YES (bundled)
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libm (mathematical functions) ______________________________ : YES
    jemalloc ___________________________________________________ : NO
    TCMalloc ___________________________________________________ : NO
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : NO
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

### Additional info

I need to run with hostNetwork: false for reasons, 

According to [the docs](https://learn.netdata.cloud/docs/installing/docker#change-the-default-hostname); I believe this should work.
>Alternatively, you can directly use the hostname from the node running the container by mounting /etc/hostname from the host in the container. With docker run, this can be done by adding --volume /etc/hostname:/etc/hostname:ro to the options. If you are using Docker Compose, you can add an entry to the container's volumes section reading - /etc/hostname:/etc/hostname:ro.

I previously tried to override the environment and use a valueFrom/FieldRef
```
child:
    env:
    - name: HOSTNAME
       valueFrom:
         fieldRef:
           apiVersion: v1
           fieldPath: spec.nodeName
``` 
however, I believe there's a separate issue in the helm chart where the env is a map `{}`, rather a list `[]`; and using this method does not work.

**Comment by ilyam8** [comment]
@bdkuhman I changed the docs to `/etc/hostname:/host/etc/hostname:ro` and added using it as a Netdata hostname in #16401 

So I think the following will work after #16401:

```yaml
    extraVolumeMounts:
        - mountPath: /host/etc/hostname
          name: hostname
    extraVolumes:
        - name: hostname
          hostPath:
            path: /etc/hostname
            type: File
            readOnly: true
```
